Ginger/GingerCore/ALM/QtestCore.cs

```diff
@@ -0,0 +1,655 @@
+        {
+            result = string.Empty;
+            if (bizFlow.ExternalID == "0" || String.IsNullOrEmpty(bizFlow.ExternalID))
+            {
+                result = GingerDicser.GetTermResValue(eTermResKey.BusinessFlow) + ": " + bizFlow.Name + " is missing ExternalID, cannot locate QC TestSet without External ID";
+                return false;
+            }
+
+            try
+            {
+                //get the BF matching test set
+                ConnectALMServer();
+                testcaseApi = new QTestApi.TestcaseApi(connObj.Configuration);
+                testrunApi = new QTestApi.TestrunApi(connObj.Configuration);
+                testlogApi = new QTestApi.TestlogApi(connObj.Configuration);
+
+                QtestTestSuite testSuite = GetQtestTestSuite(bizFlow.ExternalID);
+                if (testSuite != null)
+                {
+                    //get all BF Activities groups
+                    ObservableList<ActivitiesGroup> activGroups = bizFlow.ActivitiesGroups;                    
+                    if (activGroups.Count > 0)
+                    {
+                        foreach (ActivitiesGroup activGroup in activGroups)
+                        {
+                            if ((publishToALMConfig.FilterStatus == FilterByStatus.OnlyPassed && activGroup.RunStatus == eActivitiesGroupRunStatus.Passed)
+                            || (publishToALMConfig.FilterStatus == FilterByStatus.OnlyFailed && activGroup.RunStatus == eActivitiesGroupRunStatus.Failed)
+                            || publishToALMConfig.FilterStatus == FilterByStatus.All)
+                            {
+                                QtestTest tsTest = null;
+                                //go by TC ID = TC Instance ID
+                                tsTest = testSuite.Tests.Where(x => x.TestID == activGroup.ExternalID).FirstOrDefault();                              
+                                if (tsTest != null)
+                                {
+                                    //get activities in group
+                                    List<Activity> activities = (bizFlow.Activities.Where(x => x.ActivitiesGroupID == activGroup.Name)).Select(a => a).ToList();
+                                    string TestCaseName = PathHelper.CleanInValidPathChars(tsTest.TestName);
+                                    if ((publishToALMConfig.VariableForTCRunName == null) || (publishToALMConfig.VariableForTCRunName == string.Empty))
+                                    {
+                                        String timeStamp = DateTime.Now.ToString("dd-MMM-yyyy HH:mm:ss");
+                                        publishToALMConfig.VariableForTCRunName = "GingerRun_" + timeStamp;
+                                    }
+
+                                    //RunFactory runFactory = (RunFactory)tsTest.RunFactory;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)